src/viam/examples/mlmodel/example_audio_classification_client.cpp

```diff
@@ -210,20 +218,24 @@ int main(int argc, char* argv[]) try {
             return EXIT_FAILURE;
         }
 
-        if (!opt_robot_secret) {
+        if (!opt_api_key || !opt_api_key_id) {
             std::cout << argv[0]
-                      << ": The `--robot-secret` argument is required when connecting to a robot"
+                      << ": The `--robot-api-key` and the `--robot-api-key-id` argument are "
+                         "required when connecting to a robot"
                       << std::endl;
             return EXIT_FAILURE;
         }
 
         // Connect to the robot at the given URL with the provided
         // secret. Please see other examples for more details on
         // connecting to robots with the C++ SDK.
-        vsdk::DialOptions dial_opts;
-        dial_opts.set_credentials(vsdk::Credentials(opt_robot_secret.get()));
-        auto robot =
-            vsdk::RobotClient::at_address(opt_robot_host.get(), {0, {std::move(dial_opts)}});
+        viam::sdk::DialOptions dial_options;
+        dial_options.set_entity(opt_api_key_id.get());
+        dial_options.set_credentials(viam::sdk::Credentials("api-key", opt_api_key.get()));
+
+        auto robot = vsdk::RobotClient::at_address(
+            opt_robot_host.get(),
+            viam::sdk::Options(1, boost::optional<viam::sdk::DialOptions>(dial_options)));
```

Comment:
I'm still not clear on why the original code for these lines needed to be changed at all. As far as I can see it should have continued to work just as it was. Also, why the change from `0` to `1` for `refresh_interval`?

Can you please try restoring it to
```
auto robot =
            vsdk::RobotClient::at_address(opt_robot_host.get(), {0, {std::move(dial_opts)}});
```

And let me know what, if anything, breaks?

Reply:
that works so far as I can see - it compiles, I just still cannot run it

Reply:
I had no particular reason for the change to referesh interval